packages/fiber/src/core/utils.ts

```diff
@@ -138,7 +138,7 @@ export function dispose<TObj extends { dispose?: () => void; type?: string; [key
 // Each object in the scene carries a small LocalState descriptor
 export function prepare<T = THREE.Object3D>(object: T, state?: Partial<LocalState>) {
   const instance = object as unknown as Instance
-  if (state?.primitive || !instance.__r3f) {
+  if (!instance.__r3f) {
```

Comment:
Why was it that primitives' local state would always be overridden? What happens if an external object is re-used between multiple canvases?